src/overlays/actors/ovl_Bg_Ikana_Mirror/z_bg_ikana_mirror.c

```diff
@@ -28,99 +30,365 @@ const ActorInit Bg_Ikana_Mirror_InitVars = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 220, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 200, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+void BgIkanaMirror_SetupWait(BgIkanaMirror* this);
+void BgIkanaMirror_Wait(BgIkanaMirror* this, PlayState* play);
+void BgIkanaMirror_SetupEmitLight(BgIkanaMirror* this);
+void BgIkanaMirror_EmitLight(BgIkanaMirror* this, PlayState* play);
+
+void BgIkanaMirror_SetQuadVertices(BgIkanaMirror* this) {
+    ColliderQuadDimInit* dim;
+    ColliderQuad* collider;
+    Vec3f a;
+    Vec3f b;
+    Vec3f c;
+    Vec3f d;
+    s32 i;
+
+    Matrix_Push();
+    Matrix_SetTranslateRotateYXZ(this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y,
+                                 this->dyna.actor.world.pos.z, &this->dyna.actor.shape.rot);
+    for (i = 0; i < ARRAY_COUNT(this->colliderQuad); i++) {
+        dim = &sQuadInit[i].dim;
+        Matrix_MultVec3f(&dim->quad[0], &a);
+        Matrix_MultVec3f(&dim->quad[1], &b);
+        Matrix_MultVec3f(&dim->quad[2], &c);
+        Matrix_MultVec3f(&dim->quad[3], &d);
+        collider = &this->colliderQuad[i];
+        Collider_SetQuadVertices(collider, &a, &b, &c, &d);
+    }
+
+    Matrix_Pop();
+}
+
+void BgIkanaMirror_Init(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    BgIkanaMirror* this = THIS;
+    Vec3f* vtx;
+    Vec3f vertices[3];
+    ColliderTrisElementInit* colliderElement;
+    s32 i;
+    s32 j;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, 0);
+    DynaPolyActor_LoadMesh(play, &this->dyna, &gStoneTowerTempleMirrorCol);
+    Collider_InitTris(play, &this->colliderTris);
+    Collider_SetTris(play, &this->colliderTris, &this->dyna.actor, &sTrisInit, this->colliderTrisElements);
+    Matrix_SetTranslateRotateYXZ(this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y,
+                                 this->dyna.actor.world.pos.z, &this->dyna.actor.shape.rot);
+
+    for (i = 0; i < ARRAY_COUNT(sTrisElementsInit); i++) {
+        colliderElement = &sTrisInit.elements[i];
+        vtx = &colliderElement->dim.vtx[0];
+        for (j = 0; j < 3; j++) {
+            Matrix_MultVec3f(vtx, &vertices[j]);
+            vtx++;
+        }
+        Collider_SetTrisVertices(&this->colliderTris, i, &vertices[0], &vertices[1], &vertices[2]);
+    }
+
+    for (i = 0; i < ARRAY_COUNT(this->colliderQuad); i++) {
+        Collider_InitQuad(play, &this->colliderQuad[i]);
+        Collider_SetQuad(play, &this->colliderQuad[i], &this->dyna.actor, &sQuadInit[i]);
+    }
+
+    BgIkanaMirror_SetQuadVertices(this);
+    this->lightAbsorptionAnimatedTexture = Lib_SegmentedToVirtual(&gStoneTowerTempleMirrorLightAbsorptionTexAnim);
+    this->lightEmissionAnimatedTexture = Lib_SegmentedToVirtual(&gStoneTowerTempleMirrorLightEmissionTexAnim);
```

Comment:
Personally I don't mind the long names on the TexAnims themselves, but you can shorten the struct variables, and name them after the type of texture animation, in this case, `lightEmissionTexScroll` etc.

Reply:
Thanks! Applied in the next commit to be pushed.